app/api/events.py

```diff
@@ -93,9 +93,28 @@ def get_upcoming_events(request: Request, token: AccessTokenPayload = Depends(op
     return [Event.model_validate(event) for event in upcoming_events]
 
 
+@router.get('/past-events/count')
+def get_past_events_count(request: Request, token: AccessTokenPayload = Depends(optional_authentication)):
+    """Get count of past events"""
+    db = get_database(request)
+
+    # Get current UTC datetime
+    current_datetime = datetime.now(timezone.utc)
+
+    # Apply search filter according to role
+    search_filter = {"date": {"$lt": current_datetime}}
+    if token and token.role != Role.admin:
+        search_filter["public"] = True
+
+    # Count the number of documents that match the filter
+    count = db.events.count_documents(search_filter)
+    return {"count": count}
+
 @router.get('/past-events')
-def get_past_events(request: Request, token: AccessTokenPayload = Depends(optional_authentication)):
-    """ Get last 10 events that have passed """
+def get_past_events(request: Request, token: AccessTokenPayload = Depends(optional_authentication),
+                    skip: int = Query(0, ge=0),
+                    limit: int = Query(10,ge=1,le=50)):
```

Comment:
Like I commented on the frontend, another design here that might be simpler is to simply use a hardcoded limit, instead of allowing clients to supply it in the query parameters. If you choose to simplify it, remember to do it both on the api and the frontend.